source/includes/dev/_microservice.md

```diff
@@ -1,17 +1,109 @@
+- app.constants.js: microservice application constants.
+- app.js: `koa` bootstrap, as well as basic error handling and Control Tower registration.
+- loader.js: convenience file that iterates over the nested content of the `routes` folder and loads files.
+- logger.js: convenience file that configures the logger for ease of use.
+
+### Adding a new endpoint
+
+In this section we'll cover how you can add a new endpoint with new functionality to an existing microservice. The aim is not to be a comprehensive guide to cover all cases, but more of a quick entry point into day-to-day actions you may want to perform, which should be complemented by your own learning of how a microservice works - remember that all microservices, despite being structurally similar, have their own custom code and functionality.
+
+To add a new endpoint, here's the short tasklist you have to tackle:
+
+- Register your route in koa.
+- Add a handler for that route.
+- Add middleware for validation, if applicable.
+- Implement new services, models or serializers to handle your application logic, if applicable.
+- Add tests for your functionality (you may want to start with this, if TDD is your thing).
+- Update the Control Tower registration file.
+
+### Register your route in koa
+
+This can be done in the `app/src/routes/api/v1/dataset.router.js` file, usually at the bottom of if:
+
+```javascript
+
+// router object declaration, usually at the top of the file
+const router = new Router({
+    prefix: '/dataset',
+});
+
+// routes declaration, usually at the bottom of the file
+router.get('/', DatasetRouter.getAll);
+router.post('/find-by-ids', DatasetRouter.findByIds);
+router.post('/', validationMiddleware, authorizationMiddleware, authorizationBigQuery, DatasetRouter.create);
+// router.post('/', validationMiddleware, authorizationMiddleware, authorizationBigQuery, authorizationSubscribable, DatasetRouter.create);
+router.post('/upload', validationMiddleware, authorizationMiddleware, DatasetRouter.upload);
+router.post('/:dataset/flush', authorizationMiddleware, DatasetRouter.flushDataset);
+router.post('/:dataset/recover', authorizationRecover, DatasetRouter.recover);
+
+router.get('/:dataset', DatasetRouter.get);
+router.get('/:dataset/verification', DatasetRouter.verification);
+router.patch('/:dataset', validationMiddleware, authorizationMiddleware, DatasetRouter.update);
+router.delete('/:dataset', authorizationMiddleware, DatasetRouter.delete);
+router.post('/:dataset/clone', validationMiddleware, authorizationMiddleware, DatasetRouter.clone);
+```
+
+In here you'll find the already existing routes. As you can see from the rather explicit syntax, you need to call the method that matches the desired HTTP verb on the `router` object, and pass it a variable number of arguments - more on this in the next section. One thing to keep in mind is that all the routes in a file are typically prefixed, as defined in the `router` object declaration.
```

Comment:
I'd add here a statement saying that we use `koa-router` for the routers and a link to those docs, for more info: https://github.com/ZijianHe/koa-router